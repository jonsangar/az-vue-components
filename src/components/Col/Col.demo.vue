<template lang="pug">
  div
    Demo(title="Columnas")
      .content
        az-row(hgutter="small" grid)
          az-col(cols="1")
            .block 1
          az-col(cols="11")
            .block 11
      .content
        az-row(hgutter="small" grid)
          az-col(cols="2")
            .block 2
          az-col(cols="10")
            .block 10
      .content
        az-row(hgutter="small" grid)
          az-col(cols="3")
            .block 3
          az-col(cols="9")
            .block 9
      .content
        az-row(hgutter="small" grid)
          az-col(cols="4")
            .block 4
          az-col(cols="8")
            .block 8
      .content
        az-row(hgutter="small" grid)
          az-col(cols="5")
            .block 5
          az-col(cols="7")
            .block 7
      .content
        az-row(hgutter="small" grid)
          az-col(cols="6")
            .block 6
          az-col(cols="6")
            .block 6
      .content
        az-row(hgutter="small" grid)
          az-col(cols="7")
            .block 7
          az-col(cols="5")
            .block 5
      .content
        az-row(hgutter="small" grid)
          az-col(cols="8")
            .block 8
          az-col(cols="4")
            .block 4
      .content
        az-row(hgutter="small" grid)
          az-col(cols="9")
            .block 9
          az-col(cols="3")
            .block 3
      .content
        az-row(hgutter="small" grid)
          az-col(cols="10")
            .block 10
          az-col(cols="2")
            .block 2
      .content
        az-row(hgutter="small" grid)
          az-col(cols="11")
            .block 11
          az-col(cols="1")
            .block 1
      .content
        az-row(hgutter="small" grid)
          az-col(cols="12")
            .block 12

    Demo(title="Offset")
      .content
        az-row(hgutter="small" grid)
          az-col(cols="11" offset="1")
            .block 11

      .content
        az-row(hgutter="small" grid)
          az-col(cols="10" offset="2")
            .block 10

      .content
        az-row(hgutter="small" grid)
          az-col(cols="9" offset="3")
            .block 9

      .content
        az-row(hgutter="small" grid)
          az-col(cols="8" offset="4")
            .block 8

      .content
        az-row(hgutter="small" grid)
          az-col(cols="7" offset="5")
            .block 7

      .content
        az-row(hgutter="small" grid)
          az-col(cols="6" offset="6")
            .block 6

      .content
        az-row(hgutter="small" grid)
          az-col(cols="5" offset="7")
            .block 5

      .content
        az-row(hgutter="small" grid)
          az-col(cols="4" offset="8")
            .block 4

      .content
        az-row(hgutter="small" grid)
          az-col(cols="3" offset="9")
            .block 3

      .content
        az-row(hgutter="small" grid)
          az-col(cols="2" offset="10")
            .block 2

      .content
        az-row(hgutter="small" grid)
          az-col(cols="1" offset="11")
            .block 1

    Demo(title="Nesting")
      .content
        az-row(hgutter="small" vgutter="small" grid)
          az-col(cols="6")
            .block
              az-row(hgutter="small" vgutter="small" grid)
                az-col(cols="12")
                  .block 12
                az-col(cols="6")
                  .block 6
                az-col(cols="6")
                  .block 6
          az-col(cols="6")
            .block 4

    Demo(title="Auto / Expand")
      .content
        az-row(hgutter="small" grid)
          az-col(auto style="width: 100px")
            .block Auto
          az-col(expand)
            .block Expand

    Demo(title="Responsive")
      .content
        az-row(hgutter="small" vgutter="small" grid)
          az-col(cols="4" :palm={cols:'12'})
            .block 4
          az-col(cols="4" :palm={cols:'6'})
            .block 4
          az-col(cols="4" :palm={cols:'6'})
            .block 4

      .content
        az-row(hgutter="small" vgutter="small" grid)
          az-col(cols="4" :palm={cols:'12'})
            .block 4
          az-col(cols="4" :palm={cols:'5'})
            .block 4
          az-col(cols="4" :palm={cols:'5', offset:'2'})
            .block 4
</template>

<script>
export default {};
</script>

<style lang="scss" scoped>
.content {
  margin-bottom: half();
  overflow: hidden;
  background-color: color(gray6);
  border-radius: 3px;
}

.block {
  font-size: font-size(small2);
  font-weight: font-weight(bold);
  color: white;
  text-align: center;
  text-overflow: ellipsis;
  white-space: nowrap;
  background-color: color(brand);
  border-radius: 3px;

  .block {
    background-color: color(brand-dark);
  }
}
</style>
