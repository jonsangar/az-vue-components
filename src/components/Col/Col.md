---
device: true
---

# Col

### Col

```javascript
import { Col } from 'didor';

Vue.use(Col);
```

---

### Uso

:::note
Para que el componente funcione correctamente, es necesario definir la propiedad `grid` en el componente `az-row`.
:::

#### Columnas

```html
<az-row grid>
  <az-col cols="3">
    <div>3</div>
  </az-col>
  <az-col cols="6">
    <div>6</div>
  </az-col>
  <az-col cols="3">
    <div>3</div>
  </az-col>
</az-row>

<az-row grid>
  <az-col cols="4">
    <div>4</div>
  </az-col>
  <az-col cols="8">
    <div>8</div>
  </az-col>
</az-row>
```

#### Offset

```html
<az-row grid>
  <az-col cols="4">
    <div>4</div>
  </az-col>
  <az-col cols="3" offset="5">
    <div>3</div>
  </az-col>
</az-row>

<az-row grid>
  <az-col cols="3">
    <div>3</div>
  </az-col>
  <az-col cols="4" offset="1">
    <div>4</div>
  </az-col>
  <az-col cols="3" offset="1">
    <div>3</div>
  </az-col>
</az-row>
```


#### Nested

```html
<az-row grid>
  <az-col cols="6">
    <div>
      <az-row grid>
        <az-col cols="12">
          .block 12
        </az-col>
        <az-col cols="6">
          .block 6
        </az-col>
        <az-col cols="6">
          .block 6
        </az-col>
      </az-row>
    </div>
  </az-col>
  <az-col cols="6">
    <div>4</div>
  </az-col>
</az-row>
```

#### Auto / Expand

```html
<az-row grid>
  <az-col auto style="width: 100px">
    <div>Auto</div>
  </az-col>
  <az-col expand>
    <div>Expand</div>
  </az-col>
</az-row>
```

#### Responsive

```html
<az-row grid>
  <az-col cols="4" :palm="{cols:12}">
    <div>4</div>
  </az-col>
  <az-col cols="4" :palm="{cols:6}">
    <div>4</div>
  </az-col>
  <az-col cols="4" :palm="{cols:6}">
    <div>4</div>
  </az-col>
</az-row>

<az-row grid>
  <az-col cols="4" :palm="{cols:12}">
    <div>4</div>
  </az-col>
  <az-col cols="4" :palm="{cols:5}">
    <div>4</div>
  </az-col>
  <az-col cols="4" :palm="{cols:5, offset:2}">
    <div>4</div>
  </az-col>
</az-row>
```

---

### API

| Attribute   | Description                                | Type      | Default   |
| ----------- | ------------------------------------------ | --------- | --------- |
| cols        | Determina el número de columnas que ocupará un elemento | `Number`  | `12` |
| offset      | Determina el número de columnas de espacio que un elemento dejará a su izquierda | `Number`  | -- |
| auto        | Hace que un elemento ocupe su anchura por defecto | `Boolean` | `false`   |
| expand      | Hace que un elemento ocupe todo el espacio que queda libre en una fila | `Boolean` | `false`   |
| palm<br>lap<br>small<br>desk<br>large  | Permiten modificar algunas propiedades en función del tamaño de la pantalla, las propiedades responsive son:<br>`cols` `offset`             | `Object` | --    |
