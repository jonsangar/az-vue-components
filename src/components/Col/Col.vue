<template lang="pug">
.col(
  :class="classObject"
)
  slot
</template>

<script>
import ResponsiveClass from '../mixins/responsiveClass';

export default {
  mixins: [ResponsiveClass],
  props: {
    cols: {
      type: [String, Number],
      default: null,
    },
    auto: {
      type: Boolean,
      default: false,
    },
    expand: {
      type: Boolean,
      default: false,
    },
    offset: {
      type: [String, Number],
      default: null,
    },
    palm: { type: Object },
    lap: { type: Object },
    small: { type: Object },
    desk: { type: Object },
    large: { type: Object },
  },
  computed: {
    classObject() {
      const baseClass = 'col';
      const responsiveProperties = ['cols', 'offset'];
      const responsiveClass = this.getResponsiveClass(baseClass, responsiveProperties);

      const propertiesClass = [
        this.cols > 0 ? `col--cols-${this.cols}` : '',
        this.offset > 0 ? `col--offset-${this.offset}` : '',
        {
          'col--auto': this.auto,
          'col--expand': this.expand,
        },
      ];

      return propertiesClass.concat(responsiveClass);
    },
  },
};
</script>
