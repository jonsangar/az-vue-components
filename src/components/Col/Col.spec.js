import { shallowMount, createLocalVue } from '@vue/test-utils';
// Importamos librerías de estilos e iconos
import SvgIcon from 'vue-svgicon'
import Col from './Col.vue';

const localVue = createLocalVue()
localVue.use(SvgIcon, {
  tagName: 'az-icon',
  classPrefix: 'az',
})

describe('Col.vue', () => {
  const slotContent = '<span>foo</span>';

  const build = () => {
    const wrapper = shallowMount(Col, {
      localVue,
      slots: {
        default: slotContent
      }
    })

    return {
      wrapper,
    }
  }

  it('Comprueba renderizado del componente Col con snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Renderiza el contenido del slot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toContain(slotContent);
  });

  it('Comprueba que tiene responsive', async () => {
    const { wrapper } = build();
    // Si actualizamos la propiedad celda, añade la clase celda
    await wrapper.setProps({
      cols: '3',
      // offset: 3,
      // auto: true,
      // expand: true,
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
    expect(wrapper.html()).toMatchSnapshot()
    // Comprobamos que se ha activado la prop
    // expect(wrapper.vm.cell).toStrictEqual(true);
    // Comprobamos que se ha añadido el titulo con estilos por defecto
    // expect(wrapper.find('.az-checkbox-group--cell').isVisible()).toBe(true);
  });
});
