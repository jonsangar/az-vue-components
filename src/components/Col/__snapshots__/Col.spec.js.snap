// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col.vue Comprueba que tiene responsive 1`] = `<div class="col col--cols-3"><span>foo</span></div>`;

exports[`Col.vue Comprueba renderizado del componente Col con snapshot 1`] = `<div class="col"><span>foo</span></div>`;
